client/asset/btc/btc.go

```diff
@@ -824,25 +824,67 @@ func (btc *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
 			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount, fundingCoin.redeemScript))
 			continue
 		}
-		notFound[pt] = struct{}{}
+		notFound[pt] = true
 	}
-
 	if len(notFound) == 0 {
 		return coins, nil
 	}
+
+	// Check locked outputs for not found coins.
+	lockedOutpoints, err := btc.wallet.ListLockUnspent()
+	if err != nil {
+		return nil, err
+	}
+	for _, rpcOP := range lockedOutpoints {
+		txHash, err := chainhash.NewHashFromStr(rpcOP.TxID)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", rpcOP.TxID, err)
+		}
+		pt := newOutPoint(txHash, rpcOP.Vout)
+		if notFound[pt] {
+			txOut, err := btc.node.GetTxOut(txHash, rpcOP.Vout, true)
+			if err != nil {
+				return nil, fmt.Errorf("gettxout error for locked outpoint %v: %v", pt.String(), err)
+			}
+			var address string
+			if len(txOut.ScriptPubKey.Addresses) > 0 {
+				address = txOut.ScriptPubKey.Addresses[0]
+			}
+			pkScript, err := hex.DecodeString(txOut.ScriptPubKey.Hex)
+			if err != nil {
+				return nil, fmt.Errorf("error decoding pubkey script from hex '%s': %v",
+					txOut.ScriptPubKey.Hex, err)
+			}
+			nfo, err := dexbtc.InputInfo(pkScript, nil, btc.chainParams)
```

Comment:
It's not clear to me how `listunspent` sets a redeem script for returned outputs. This will error for P2SH and P2WSH outputs. Appreciate any pointers I can get before I find the time to look deeper.

Reply:
Yes, it's a problem after all.  The redeem script is required to know how big the input in the spending txn will be, and this is even sent to the server with the order so it can verify the funding coin amounts are sufficient for txn fees.  While I think this only is important for FundOrder, which is used when placing new orders like this, and probably not FundingCoins, which is used to regenerate the asset.Coins for existing orders, I'm wary of hacking around this.

Since there seems no other way to get the redeem scripts other than listunspent, it appears that the solution is to unlock the locked order coins, then do listunspent, then use FundingCoins.

The above is a summary of our chat in matrix.

Reply:
t's pretty unfortunate to have to do it this way though since I'm pretty certain that these redeem scripts are only required when submitting the order and funding adequately, but that's all done.  In FundingCoins we're just recovering asset.Coins for existing orders...  @buck54321, thoughts about this?